packages/build-tools/src/utils/outputs.ts

```diff
@@ -0,0 +1,80 @@
+import { Generic } from '@expo/eas-build-job';
+import { BuildStep, jsepEval } from '@expo/steps';
+import { bunyan } from '@expo/logger';
+import nullthrows from 'nullthrows';
+import fetch from 'node-fetch';
+
+import { BuildContext } from '../context';
+
+export async function uploadJobOutputsToWwwAsync(
+  ctx: BuildContext<Generic.Job>,
+  {
+    steps,
+    logger,
+    expoApiV2BaseUrl,
+  }: { steps: BuildStep[]; logger: bunyan; expoApiV2BaseUrl: string }
+): Promise<void> {
+  if (!ctx.job.outputs) {
+    logger.info('Job defines no outputs, skipping upload');
+    return;
+  }
+
+  try {
+    const workflowJobId = nullthrows(ctx.job.builderEnvironment?.env?.__WORKFLOW_JOB_ID);
+    const robotAccessToken = nullthrows(ctx.job.secrets?.robotAccessToken);
+
+    const interpolationContext = {
+      steps: Object.fromEntries(steps.map((step) => [step.id, getDynamicValuesFromStep(step)])),
+    };
+    logger.info({ dynamicValues: interpolationContext }, 'Using dynamic values');
+
+    const outputs = getJobOutputsFromSteps({
+      jobOutputDefinitions: ctx.job.outputs,
+      interpolationContext,
+    });
+    logger.info('Uploading outputs');
+
+    await fetch(new URL(`workflows/${workflowJobId}`, expoApiV2BaseUrl).toString(), {
```

Comment:
we can add a util like `turtleFetch` here if we want to:
https://github.com/expo/turtle-v2/blob/998063fdfb9edb61eba3e080a960a0de067139ab/src/libs/turtle-common/src/turtleFetch.ts#L16-L74

Reply:
It *used to* use `turtleFetch` when it was located in turtle, but it's unavailable here so I replace it with the standard `fetch`, but you're right that we need something similar, if only just for retries